### PR TITLE
Add support to painless sort script

### PR DESCRIPTION
**Pull request by pereirapleandro**
- [x] Remove default sort
- [x] Add support to sort by painless script
- [x] Resolve deprecateds
- [x] Upgrade ES version to `6.4.2`
- [x] Add healthcheck to docker-compose
- [x] Remove `cursorId` from response
- [x] Remove class `EsClient` to use `RestClient` to get cluster information

Close https://github.com/grupozap/squad-search-ranking/issues/1175

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -17,7 +17,7 @@ buildscript {
         classpath 'com.avast.gradle:docker-compose-gradle-plugin:0.5.0'
         classpath 'org.sonarsource.scanner.gradle:sonarqube-gradle-plugin:2.6.2'
         classpath 'com.github.ben-manes:gradle-versions-plugin:+'
-        classpath 'com.gradle:build-scan-plugin:+'
+        classpath 'com.gradle:build-scan-plugin:1.16'
```
Comment (mezzatto) [inline build.gradle]: `1.+` ?

File: src/integration-test/resources/docker/docker-compose-circleci.yml [hunk 2]
```diff
@@ -18,8 +19,14 @@ services:
     ports:
       - "9200:9200"
       - "9300:9300"
+    healthcheck:
+      test: "curl http://localhost:9200/_cat/health | grep green || exit 1"
```
Comment (mezzatto) [inline src/integration-test/resources/docker/docker-compose-circleci.yml]: 🏆

File: src/integration-test/resources/docker/docker-compose-circleci.yml [hunk 3]
```diff
@@ -46,3 +54,8 @@ services:
     ports:
       - "5005:5005"
       - "8482:8482"
+    healthcheck:
+      test: "wget --spider -S http://localhost:8482/actuator/health 2>&1 | grep 'HTTP/' | awk '{print $$2}' | grep 200 || exit 1"
```
Comment (mezzatto) [inline src/integration-test/resources/docker/docker-compose-circleci.yml]: 🏆

File: build.gradle (hunk 1 above)
Comment (pereirapleandro) [reply build.gradle]: great!

File: src/main/java/com/grupozap/search/api/model/parser/FacetParser.java [hunk 4]
```diff
@@ -30,7 +30,7 @@ public FacetParser(FieldParser fieldParser, SortParser sortParser) {
             (void1, void2, sort) -> sort);
 
     this.facetParser =
-        sequence(fieldParser.getWithoutNot(), sortFacetParser.asOptional(), Facet::new)
+        sequence(fieldParser.getWithNot(), sortFacetParser.asOptional(), Facet::new)
```
Comment (mezzatto) [inline src/main/java/com/grupozap/search/api/model/parser/FacetParser.java]: Could you please explain this change? Was there a bug in here?

**Comment by mezzatto** [comment]
I approved so that I'm not a blocker. To be honest, I did not understand the whole code. It has too many changes of too many different unrelated scopes.

Let's trust the tests :D

**Pull request merged**